keras_nlp/models/whisper/whisper_tokenizer.py

```diff
@@ -0,0 +1,192 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Whisper tokenizer."""
+
+import json
+
+from keras_nlp.api_export import keras_nlp_export
+from keras_nlp.tokenizers.byte_pair_tokenizer import BytePairTokenizer
+
+
+def _load_dict(dict_or_path):
+    if isinstance(dict_or_path, str):
+        with open(dict_or_path, "r") as f:
+            dict_or_path = json.load(f)
+    return dict_or_path
+
+
+@keras_nlp_export("keras_nlp.models.WhisperTokenizer")
+class WhisperTokenizer(BytePairTokenizer):
+    """Whisper text tokenizer using Byte-Pair Encoding subword segmentation.
+
+    This tokenizer class will tokenize raw strings into integer sequences and
+    is based on `keras_nlp.tokenizers.BytePairTokenizer`.
+    This tokenizer does not provide truncation or padding of inputs.
+
+    Args:
+        vocabulary: string or dict, maps token to integer ids. If it is a
+            string, it should be the file path to a json file.
+        merges: string or list, contains the merge rule. If it is a string,
+            it should be the file path to merge rules. The merge rule file
+            should have one merge rule per line. Every merge rule contains
+            merge entities separated by a space.
+        special_tokens: string or dict, maps special tokens to integer IDs. If
+            it is a string, it should be the path to a JSON file.
+        language_tokens: string or dict, maps language tokens to integer IDs. If
+            non-None or empty, the tokenizer will be assumed to be a
```

Comment:
let's just say `if not None`, why give people two ways to specify a mono-lingual tokenizer?